### PR TITLE
WIP: Use globby gitignore handling

### LINKED ISSUE #236: Move the gitignore handling code to `globby`
As an optional feature in `globby`. It was a really difficult thing to implement, so would be nice if everyone could take advantage of and help improve it.

@marionebl @schnittstabil Thoughts?

### PR DESCRIPTION
**Pull request by marionebl**
* closes #236
* ~~failing test cases indicate regression in globby gitignore handling~~ Just faulty usage of globby

## ToDo

* [x] expose `getGitIgnoreFilter` from globby: [globby#56](https://github.com/sindresorhus/globby/pull/56)
* [x] investigate failing test cases
* [x] remove all remnants of `lib/gitignore`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -87,13 +87,19 @@ exports.lintText = (str, opts) => {
 exports.lintFiles = (patterns, opts) => {
 	opts = optionsManager.preprocess(opts);
 
+	const globbyOptions = {
+		cwd: opts.cwd,
+		gitignore: true,
+		ignore: opts.ignores,
+		nodir: true
```
Comment (sindresorhus) [inline index.js]: `nodir` is already `true` by default in Globby.

**Comment by sindresorhus** [comment]
Generally looks good, but some of the tests seems to be failing.

**Comment by marionebl** [comment]
Will pick this up again in the course of the next two weeks

**Comment by pvdlg** [comment]
@marionebl are your still working on this PR? I can try to pick it up if you'd like.

**Comment by marionebl** [comment]
@pvdlg Sure, go ahead!

**Comment by marionebl** [comment]
Closing in favor of #283

**Pull request closed without merging**